controllers/namespacetarget/target.go

```diff
@@ -32,7 +32,23 @@ type NamespaceTarget struct {
 var _ bindings.SecretDeploymentTarget = (*NamespaceTarget)(nil)
 
 func (t *NamespaceTarget) GetSpec() api.LinkableSecretSpec {
-	return *t.SecretSpec
+	ret := t.SecretSpec.DeepCopy()
+	if t.TargetSpec.Secret.Name != "" {
+		ret.Name = t.TargetSpec.Secret.Name
+	}
+	if t.TargetSpec.Secret.GenerateName != "" {
```

Comment:
Do we really need generated name?
One of the point of such spec extending is predictability, so I would consider the one who create such a target usually wants to care about its name.